pyro/infer/autoguide/effect.py

```diff
@@ -312,7 +313,8 @@ def _get_params(self, name: str, prior: Distribution):
                 init_weight = torch.full_like(init_loc, self._init_weight)
             # if site is hierarchical substract contribution of dependencies from init_loc
             if (self._hierarchical_sites is None) or (name in self._hierarchical_sites):
-                init_loc = init_loc - init_weight * transform.inv(prior.mean)
+                init_prior_mean = self._adjust_plates(prior.mean, event_dim)
+                init_loc = init_loc - init_weight * transform.inv(init_prior_mean)
```

Comment:
you'll need to transform before applying `self._adjust_plates` because the transform could introduce batch dimensions e.g. for an `AffineTransform`